### ISSUE TITLE
Feature Request: ability to merge new rules instead of overriding

### ISSUE THREAD

**Issue by sfabriece** [palantir/tslint#3474]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.4.2
- __Running TSLint via__: CLI 

#### Request

currently base rules get overridden, it is disirable to be able to merge arguments instead of overriding.

A simple use case is a rule that has common arguments, but where you want consumers to be able to extend it to add more arguments

**Comment by ajafff** [comment]
I agree that there are valid use cases. Before we implement such a feature, we need a proposal how it should work. It should be easily understandable even by people new to tslint.

**Comment by sfabriece** [comment]
Is this something that I have to do? I'm not very familiar with the internals of tslint. But happy to be involved so that I can learn.

**Comment by ajafff** [comment]
@sfabriece This is not something you have to come up with. Since this is open source, everyone can contribute.

Also we are not talking about code (yet). First we need to decide how the config file should look like.

In some project I saw the keys prefixed with `+` to add it to the existing config instead of overriding.

```js
// base.json
{
  "rules": {
    "whitespace": [true, "check-decl", "check-separator"]
  }
}
// extending.json
{
  "extends": "./base.json",
  "rules": {
    "whitespace": {
      "+options": "check-module", 
      "-options": "check-decl"
    } // results in 'whitespace: [true, "check-separator", "check-whitespace"]'
  }
}
```

Unfortunately it's not always as simple as concatenating. Some rules use an object as configuration such as https://palantir.github.io/tslint/rules/no-shadowed-variable/. In this case we could merge the objects.

The worst case is the new config of `max-line-length`. You can either use `"max-line-length": [true, 120]` or `"max-line-length": [true, {"limit": 120}]`. Don't know what we would do about that.

**Comment by ajafff** [comment]
@sarink in the other issue you said that ESLint merges configurations instead of overriding. Can you please provide a link to the documentation of that feature?

**Comment by sarink** [comment]
@ajafff I misunderstood, it behaves the same way tslint does: that is, if only the severity is specified it will be used along with the parents options, if options are specified then they will completely override the parents options (although, tslint is still not working this way for me, hopefully I'll have some time to isolate a test case to prove this)

**Comment by Chowarmaan** [comment]
In the object case, it would be nice to be able to remove one option.  For instance, a number of rules have 5 or 6 options that can be set.  Using the bang (not, !) or something to remove just the one setting, instead of having to set all the others, especially if they were extended, would be nice.

For instance, if you extend the tslint-microsoft-contrib, there are a number of options added with many object parameters.  It would be nice to take one and instruct tslint to ignore just one of the objects.  

I am trying to do this with the whitespace setting to allow my `abc:string` instead of `abc : string` and would like to simply do a "whitespace": [ !"check-type" ] or something, since there are 10 possible object parameters.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
